src/stream.js

```diff
@@ -3,5 +3,97 @@
 const logger = require('./logger');
 
 module.exports.handler = async (event) => {
+	event = decodeEvent({event});
```

Comment:
I'm pretty sure this code does not consider events from multiple channels. It's possible a single batch of events from the stream will contain events from two or more channels. We need to partition the events based on their channel first and then process each channel's events independently.

You could use something like this:
https://lodash.com/docs/4.17.15#groupBy

I'm not sure if you've ever used lodash before. If not, it is a library with lots of useful helper functions and goodies. We use it frequently with arrays/collections.

Reply:
It actually does handle that :D

Reply:
It will separate each event by date and channel.

Reply:
I haven't used `lodash` it is taboo in the frontend world.

Reply:
This is what I was thinking of how to use it: https://github.com/jkehres/trivial-slack-bot/blob/maya/sendMessageToSlack/src/stream.js#L13

Reply:
OK. I missed that then.

Lodash is a large (size-wise) dependency so I'm guessing that's why it might be taboo in the frontend. I think it does have an npm package per function, though, to deal with this issue. For example:

https://www.npmjs.com/package/lodash.groupby
https://github.com/lodash/lodash/tree/4.6.0-npm-packages

Reply:
Yeah, if you watch the ignite session - it is because of how big it is.